### PR TITLE
WOS harvest job

### LINKED ISSUE #639: Replace SW-API for WOS-API in author harvest job
See https://github.com/sul-dlss/sul_pub/blob/master/app/jobs/author_harvest_job.rb

```
$ git grep 'AuthorHarvestJob'
app/controllers/authors_controller.rb:7:    if AuthorHarvestJob.perform_later(author_params[:cap_profile_id], harvest_alternate_names: alt_names)
app/jobs/author_harvest_job.rb:1:class AuthorHarvestJob < ActiveJob::Base
app/jobs/author_harvest_job.rb:19:    msg = "AuthorHarvestJob.perform(cap_profile_id=#{cap_profile_id}, harvest_alternate_names=#{harvest_alternate_names})"
spec/controllers/authors_controller_spec.rb:17:      it 'enqueues an AuthorHarvestJob with an Author' do
spec/controllers/authors_controller_spec.rb:22:        end.to have_enqueued_job(AuthorHarvestJob).with(author.cap_profile_id.to_s, harvest_alternate_names: false)
spec/controllers/authors_controller_spec.rb:24:      it 'enqueues an AuthorHarvestJob with an Author and altNames parameter' do
spec/controllers/authors_controller_spec.rb:29:        end.to have_enqueued_job(AuthorHarvestJob).with(author.cap_profile_id.to_s, harvest_alternate_names: true)
```

(Also, what backend jobs system is this using?)

### PR DESCRIPTION
**Pull request by darrenleeweber**
Fix #639 

- added active job specs
  - consolidated common author details into a factory
- extract method refactor for the sciencewire harvest
  - use settings to enable/disable it
- add harvest method for the WOS harvester
  - extract method refactor for logging publication info

### CODE REVIEW

File: app/jobs/author_harvest_job.rb [hunk 1]
```diff
@@ -1,24 +1,52 @@
...
-    harvester.harvest_pubs_for_author_ids author.id
-    pubs = Contribution.where(author_id: author.id).map(&:publication).each do |p|
-      logger.info "publication #{p.id}: #{p.pub_hash[:apa_citation]}"
-    end
-    logger.info "Number of publications #{pubs.count}"
+    raise "Could not find or fetch author: #{cap_profile_id}" unless author.is_a?(Author)
+    sciencewire(author, harvest_alternate_names)
+    web_of_science(author, harvest_alternate_names)
   rescue => e
     msg = "AuthorHarvestJob.perform(cap_profile_id=#{cap_profile_id}, harvest_alternate_names=#{harvest_alternate_names})"
```
Comment (atz) [inline app/jobs/author_harvest_job.rb]: I know you didn't touch this line, but I think the log message intends to have `harvest_alternate_names:`, not `harvest_alternate_names=`...

File: spec/factories/author.rb [hunk 2]
```diff
@@ -1,12 +1,19 @@
 FactoryBot.define do
   sequence(:random_id) do |n|
-    @random_ids ||= (10_000..1_000_000).to_a.shuffle
+    @random_ids ||= Array.new(10_000) { SecureRandom.random_number(990_000) + 10_000 }
```
Comment (atz) [inline spec/factories/author.rb]: This factory is ridiculous.  It generates 10,000 random numbers to give back one.  Previously, it initialized a `990,000` member `Array` and shuffled it, which is **worse**!  The most authors our tests ever use is `5`.  

Why not just generate the ids on demand (per `n`), record each generated ids in a Hash (`@random_ids`) and regenerate if there is ever collision?  The only downside of this approach would be when the namespace started to fill up and we had collisions, but clearly our usage will never approach that!

File: spec/factories/author.rb [hunk 3]
```diff
@@ -32,4 +39,37 @@
       end
     end
   end
+
+  # Public data from
+  # - https://stanfordwho.stanford.edu
+  # - https://med.stanford.edu/profiles/russ-altman
+  factory :russ_altman, parent: :author do
```
Comment (atz) [inline spec/factories/author.rb]: Does this mean `author.local.rb` can go away?

File: app/jobs/author_harvest_job.rb (hunk 1 above)
Comment (darrenleeweber) [reply app/jobs/author_harvest_job.rb]: Yea, it's a bit weird for a log message to appear to construct the call, as if you could cop-n-pasta to run it again.  If that's the intention, it _could_ be fixed up.

File: spec/factories/author.rb (hunk 3 above)
Comment (darrenleeweber) [reply spec/factories/author.rb]: It's all public information; if it's also in that local file, yea, will throw that out in this commit.

File: spec/factories/author.rb (hunk 2 above)
Comment (darrenleeweber) [reply spec/factories/author.rb]: The only question with adopting a Hash solution is whether that could interfere with the design of the `sequence` method.  Originally, this was a cop-n-pasta solution and it works, although I also don't love it for these reasons ^^.  I'll do a bit more homework on that `sequence` method and see if a Hash solution works out OK.

File: spec/factories/author.rb (hunk 3 above)
Comment (darrenleeweber) [reply spec/factories/author.rb]: Turns out the local version adds a ton of publications that are specific to SW.  Until we remove SW, punting on removing that data.  This PR allows the public information to be used generally in specs and CI.

File: spec/factories/author.rb [hunk 4]
```diff
@@ -1,12 +1,33 @@
 FactoryBot.define do
   sequence(:random_id) do |n|
-    @random_ids ||= (10_000..1_000_000).to_a.shuffle
-    @random_ids[n]
+    @random_ids ||= {}
+    @random_ids[n] ||= begin
+      x = nil
+      while x.nil?
+        i = SecureRandom.random_number(990_000) + 10_000
+        x = i unless @random_ids.values.include?(i)
```
Comment (atz) [inline spec/factories/author.rb]: This is a more reasonable approach, why not just go ahead and just put the value in as the hash as the key.  That's all we care about, not `n`.

File: spec/factories/author.rb (hunk 3 above)
Comment (atz) [reply spec/factories/author.rb]: But now, the only thing we are keeping "private" is a bunch of ID numbers?  I'm not saying they need to be the same factory, but I don't see any privacy concern over a few hundred numerals.

**Pull request merged**